### ISSUE TITLE
Keras Model.compile() exception text is unclear when a loss method is not specified during compile()

### ISSUE THREAD

**Issue by drjubbs** [tensorflow/tensorflow#44942]
**System information**
- TensorFlow version (you are using): 2.3.1
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.** User is not given an obvious error message when  tensorflow.keras.models.Sequential.compile() is called and a loss function is omitted.

```
# Build the model
tf_model = Sequential()
tf_model.add(LSTM(
    units=32,
    input_shape=[window, 1],    
))
tf_model.add(Dense(units=1))
tf_model.compile()
tf_model.fit(rolling_x, rolling_y, epochs=100)
```

Triggers the following exception:

```
\env\lib\site-packages\tensorflow\python\keras\optimizer_v2\optimizer_v2.py:1270 _filter_grads
        raise ValueError("No gradients provided for any variable: %s." %
```

**Will this change the current api? How?** No

**Who will benefit with this feature?** New users

**Comment by bhack** [comment]
Can you try with `tf-nighlty`?

**Comment by drjubbs** [comment]
Yes, this still occurs with nightly. PS. Due to Nvidia software mismatch this was run on CPU and not GPU. I don't think that matters.

**Comment by bhack** [comment]
Can you fill your variable in you code example so that I could run your same code quickly?
